File: src/TrussOptimization.Web/TrussOptimizationClientApp/src/core/truss/truss-structure.ts
import { TrussLoad } from "./truss-load";
import { TrussElement } from "./truss-element";
import { Point } from "./point";
export class TrussStructure {
  topChord: TrussElement[];
  bottomChord: TrussElement[];
  bracing: TrussElement[];
  loads: TrussLoad[];
  support1: Point;
  support2: Point;

  constructor(
    topChord: TrussElement[],
    bottomChord: TrussElement[],
    bracing: TrussElement[],
    loads: TrussLoad[],
    support1: Point,
    support2: Point
  ) {
    this.topChord = topChord;
    this.bottomChord = bottomChord;
    this.bracing = bracing;
    this.loads = loads;
    this.support1 = support1;
    this.support2 = support2;
  }
}

File: src/TrussOptimization.Domain/Truss/TrussStructure.cs
using StruCal.TrussOptimization.Domain.Elements;
using StruCal.TrussOptimization.Domain.Loads;

namespace StruCal.TrussOptimization.Domain.Truss
{
    public class TrussStructure
    {
        public ElementCollection TopChord { get; }
        public ElementCollection BottomChord { get; }
        public ElementCollection Bracing { get; }

        public LoadCollection LoadCollection { get; }
        public SupportCollection SupportCollection { get; }

        public TrussStructure(ElementCollection topChord, ElementCollection bottomChord, ElementCollection bracing, LoadCollection loadCollection, SupportCollection supportCollection)
        {
            TopChord = topChord ?? throw new ArgumentNullException(nameof(topChord));
            BottomChord = bottomChord ?? throw new ArgumentNullException(nameof(bottomChord));
            Bracing = bracing ?? throw new ArgumentNullException(nameof(bracing));
            LoadCollection = loadCollection ?? throw new ArgumentNullException(nameof(loadCollection));
            SupportCollection = supportCollection ?? throw new ArgumentNullException(nameof(supportCollection));
        }
    }
}